src/components/ProgressBar/ProgressBar.vue

```diff
@@ -0,0 +1,129 @@
+<!--
+  - @copyright Copyright (c) 2020 Marco Ambrosini <marcoambrosini@pm.me>
+  -
+  - @author Marco Ambrosini <marcoambrosini@pm.me>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<docs>
+This is a simple progress bar component.
+## Usage:
+
+### Small
+<ProgressBar :value="60" />
+
+### Medium
+<ProgressBar :value="60" size="medium" />
+
+### Warn
+<ProgressBar :value="60" :warn="true" />
+
+</docs>
+
+<template>
+	<progress class="progress-bar vue"
+		:class="{ 'progress-bar--warn': warn }"
+		:style="progressBarStyle"
+		:value="value"
+		max="100" />
+</template>
+
+<script>
+export default {
+
+	name: 'ProgressBar',
+
+	props: {
+		/**
+		 * An integer between 1 and 100
+		 */
+		value: {
+			type: Number,
+			default: 0,
+		},
+		/**
+		 * Determines the height of the progressbar.
+		 * Possible values:
+		 * - 'small' (default)
+		 * - 'medium'
+		 */
+		size: {
+			type: String,
+			default: 'small',
+		},
+		/**
+		 * Applies an error color to the progressbar if true.
+		 */
+		warn: {
+			type: Boolean,
+			default: false,
+		},
+	},
+
+	computed: {
+
+		progressBarStyle() {
+			let height = 0
+			let borderRadius = 0
+			if (this.size === 'small') {
+				height = '4px'
+				borderRadius = '2px'
+			} else if (this.size === 'medium') {
+				height = '6px'
+				borderRadius = '3px'
+			}
+			return {
+				height,
+				'border-radius': borderRadius,
+			}
+		},
+	},
+}
+
+</script>
+
+<style lang="scss" scoped>
+
+.progress-bar {
+	display: block;
+	width: 100%;
+	background: var(--color-background-dark);
+	border: 0;
+	padding: 0;
+	border-radius: var(--border-radius);
+	&::-webkit-progress-bar {
+		height: calc(var(--border-radius) * 2);
+	}
+	&::-webkit-progress-value {
+		background: linear-gradient(40deg, var(--color-primary-element) 0%, var(--color-primary-element-light) 100%);
+		border-radius: var(--border-radius);
+	}
+	&::-moz-progress-bar {
+		background: linear-gradient(40deg, var(--color-primary-element) 0%, var(--color-primary-element-light) 100%);
+	}
+	&--warn {
+		// Override previous values
+		&::-moz-progress-bar {
+			background: var(--color-error) !important;
```

Comment:
So warn uses error color? Maybe change to error class and prop instead?

Reply:
Error class applied to the progress element?

Reply:
Well the warn attribute should use the color-warn. If you use color-error the controling prop should be error?

Reply:
Otherwise if at some point we want color-warn it gets really complicated